overlord/servicestate/quota_control.go

```diff
@@ -172,10 +171,38 @@ func RemoveQuota(st *state.State, name string) error {
 		return fmt.Errorf("cannot remove quota group with sub-groups, remove the sub-groups first")
 	}
 
+	// if this group has a parent, we need to remove the linkage to this
+	// sub-group from the parent first
+	if grp.ParentGroup != "" {
+		parent, ok := allGrps[grp.ParentGroup]
+		if !ok {
+			return fmt.Errorf("internal error: parent group of %q, %q not found", name, grp.ParentGroup)
+		}
+
+		newSubgroups := make([]string, 0, len(parent.SubGroups)-1)
+		for _, sub := range parent.SubGroups {
+			if sub != name {
+				newSubgroups = append(newSubgroups, sub)
```

Comment:
we don't seem to have a test that exercises this putting back other subgroups

Reply:
added